src/v/cloud_storage/partition_manifest.cc

```diff
@@ -225,7 +226,37 @@ partition_manifest::partition_manifest(
       partition_mem_tracker ? partition_mem_tracker->create_child("manifest")
                             : ss::make_shared<util::mem_tracker>(""))
   , _segments(util::mem_tracked::map<absl::btree_map, key, value>(_mem_tracker))
-  , _last_offset(0) {}
+  , _last_offset(0)
+  , _serialization_sem(1) {}
+
+partition_manifest::partition_manifest(const partition_manifest& other)
+  : _ntp(other._ntp)
+  , _rev(other._rev)
+  , _mem_tracker(other._mem_tracker)
+  , _segments(other._segments)
+  , _replaced(other._replaced)
+  , _last_offset(other._last_offset)
+  , _start_offset(other._start_offset)
+  , _last_uploaded_compacted_offset(other._last_uploaded_compacted_offset)
+  , _insync_offset(other._insync_offset)
+  , _serialization_sem(1) {}
```

Comment:
a copyable type with a mutex in it is unusual. perhaps the mutex should be located outside of the manifest?

Reply:
I tend to agree. That copy c-tor and assignment operator are only used by tests.